crates/oxc_transformer/src/es2015/shorthand_properties.rs

```diff
@@ -0,0 +1,53 @@
+use oxc_ast::{ast::*, AstBuilder};
+use oxc_span::GetSpan;
+
+use std::rc::Rc;
+
+/// ES2015: Shorthand Properties
+///
+/// References:
+/// * <https://babel.dev/docs/babel-plugin-transform-shorthand-properties>
+/// * <https://github.com/babel/babel/blob/main/packages/babel-plugin-transform-shorthand-properties>
+pub struct ShorthandProperties<'a> {
+    ast: Rc<AstBuilder<'a>>,
+}
+
+impl<'a> ShorthandProperties<'a> {
+    pub fn new(ast: Rc<AstBuilder<'a>>) -> Self {
+        Self { ast }
+    }
+
+    pub fn transform_object_property<'b>(&mut self, obj_prop: &'b mut ObjectProperty<'a>) {
+        if !obj_prop.shorthand && !obj_prop.method {
+            return;
+        }
+
+        obj_prop.shorthand = false;
```

Comment:
It seems weird that we mutate the value here then return early below in `if obj_prop.computed`?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#prototype_setter

> A property definition of the form `__proto__`: value or "`__proto__`": value does not create a property with the name `__proto__`. Instead, if the provided value is an object or [null](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/null), it points the [[Prototype]] of the created object to that value. (If the value is not an object or null, the object is not changed.)
>
> Note that the `__proto__` key is standardized syntax, in contrast to the non-standard and non-performant [`Object.prototype.__proto__`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/proto) accessors. It sets the [[Prototype]] during object creation, similar to [Object.create](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/create) — instead of mutating the prototype chain.

This means that the following code:

```js
const obj = {
  __proto__,
};
```

is not `Prototype` setter. When transpile to ES5, it will be:

```js
var obj = {
  __proto__: __proto__,
};
```

This is not correct, the semmantic is different. We should fix it to:

```js
var obj = {
  ["__proto__"]: __proto__,
};
```

another example:

```js
const obj = {
  __proto__() {},
};
```

should be transpiled to:

```js
var obj = {
  ["__proto__"]: function () {},
};
```

But there are some cases(computed key) that we can leave it as it is,

```js
const obj = {
  [__proto__]() {},
  ["__proto__"]() {},
};
```

transpiled code:

```js
var obj = {
  [__proto__]: function () {}, // not related to prototype
  ["__proto__"]: function () {}, // is already correct
};
```

Reply:
Shall we inline these cases as comments for future references?